R/summarize_over_group_and_time_range.R

```diff
@@ -0,0 +1,92 @@
+# This function summarizes data_all by grouping based on a defined category
+# (e.g. income, continent, etc) first and then summarizes over a time range.
+# Here we take average once.
+# This seems to provide better results when compared to summarizing over time
+# first, and later over group (taking averages twice)
+# 
+#' @export
+summarize_over_group_and_time_range <- function (data_all, country_last_update_info, start_date, end_date, grouped = FALSE, category = "income") {
+  
+  data_filtered <-
+    data_all |>
+    select(-ends_with("_orig")) |>
+    filter(set == "country") |>
+    select(-name, -set) |>
+    left_join(country_last_update_info, by = "unit") |>
+    filter(!is.na(last_update))|>
+    filter(dplyr::between(
+      time,
+      as.Date(start_date),
+      as.Date(end_date)
+    ))
+# browser()
+  data_summarized <-
+    data_filtered |>
+    group_by(unit) |>
+    summarize(
+      name = name[1],
+      # group = group[1],
+      pop_100k                 = mean(pop_100k, na.rm = TRUE),
```

Comment:
@christophsax @benubah why do we neeb the mean of the population?
Can we replace either by adding pop100K and pop in the group_by?
Or by adding: 
```{r} 
pop_100K = pop_100K[1],
pop = pop[1]

```

Reply:
It's a constant, so this should be equivalent. Will try to put all aggregations in a single function with a reasonable documentation. Let's review this afterward.